bindings/go/src/ucx/context_params.go

```diff
@@ -55,7 +59,13 @@ func (p *UcpParams) SetEstimatedNumPPN(estimatedNumPPN uint64) *UcpParams {
 
 // Tracing and analysis tools can identify the context using this name.
 func (p *UcpParams) SetName(name string) *UcpParams {
+	if p.params.name != nil {
+		runtime.SetFinalizer(p, nil)
+		FreeNativeMemory(unsafe.Pointer(p.params.name))
+	}
```

Comment:
1. where was this memory allocated?
2. maybe add common function for this logic?

Reply:
1. `p.params.name = C.CString(name)`
> // Go string to C string
// The C string is allocated in the C heap using malloc.
// It is the caller's responsibility to arrange for it to be
// freed, such as by calling C.free (be sure to include stdlib.h
// if C.free is needed).
func C.CString(string) *C.char
2. Ok

Reply:
This if is for cases where params are reused:
```
ucpParams.SetName("Context1")
...
ucpParams.SetName("Context2")
```